superset/db_engine_specs/base.py

```diff
@@ -1038,3 +1038,8 @@ def get_extra_params(database: "Database") -> Dict[str, Any]:
                 logger.error(ex)
                 raise ex
         return extra
+
+    @classmethod
+    def is_readonly(cls, parsed_query: ParsedQuery) -> bool:
+        """Pessimistic readonly, 100% sure statement won't mutate anything"""
+        return parsed_query.is_select() or parsed_query.is_explain()
```

Comment:
As this is moved to a more generic context, it might be a good idea to call this method `is_readonly_query`.

Reply:
good idea